### ISSUE TITLE
copy() on Windows desktop fails (<3.12)

### ISSUE THREAD

**Issue by beefeater7** [hluk/CopyQ#1440]
**Describe the bug**
Running command with `copy()` will unfocus the desktop before copying (sometimes moving focus to some window), throwing exception.

**To Reproduce**
1. Go to desktop
2. Select some files
3. Run this:
    ```
    [Command]
    Name=Copy
    Command="
        copyq:
        copy()
    "
    IsGlobalShortcut=true
    Icon=\xf15b
    GlobalShortcut=shift+f4
    ```
**Expected behavior**
Work like in Explorer or any window.

**Version, OS and Environment**
Discovered:
```
CopyQ Clipboard Manager v3.12.0-1-g9d084113
Qt: 5.13.2
Compiler: GCC
Arch: x86_64-little_endian-llp64
OS: Windows 10 (10.0)
```
Earliest known occurence:
```
CopyQ Clipboard Manager v3.11.0
Qt: 5.13.2
Compiler: GCC
Arch: i386-little_endian-ilp32
OS: Windows 10 (10.0)
```

**Comment by hluk** [comment]
Is this still an issue? I cannot reproduce it with the command on Windows 10 and CopyQ v3.13.0.

**Comment by beefeater7** [comment]
Thanks for taking a look, I'll try to reproduce it also.

Den tor 29 okt. 2020 12:58Lukas Holecek <notifications@github.com> skrev:

**Comment by beefeater7** [comment]
Yep, I still get this. It seems to relate to windows being open while the desktop is active.
```
CopyQ Clipboard Manager v3.13.0
Qt: 5.13.2
Compiler: GCC
Arch: i386-little_endian-ilp32
OS: Windows 10 (10.0)
```

**Comment by Chealer** [comment]
@beefeater7: Does this persist with version 13 or ulterior? If so, I guess the title‘s parenthesis should be removed?
Also, do you have evidence that this is specific to Windows? Ticket #1481 apparently reports the same on GNU/Linux. It does not mention "being on the desktop" as a requirement, but it also reports it is intermittent.

<details>
  <summary>License</summary>
This comment and any other from myself on this ticket is offered under the terms of CC0 1.0.
</details>

**Comment by beefeater7** [comment]
The issue persists, so I removed the (). It happens on the Windows desktop. By which I mean, every window is inactive and arrow keys navigate the items on the desktop. You can reproduce it by

1. Opening a folder in windowed Explorer
2. Selecting an item A in it
3. Keep some desktop item B visible: select it (A should remain selected in Explorer)
4. Run the command.

Explorer is refocused and A copied, not B. I'm not sure how this is supposed to be related to ticket 1481.

**Comment by Chealer** [comment]
Thank you @beefeater7 

Could you specify which CopyQ and Windows versions you last reproduced this with?

My understanding of what @beefeater7 reports follow.

### Steps
1. Open a folder in File Explorer
2. Select a file (A) in it
3. Select a file (B) on the desktop, while the File Explorer window remains open and the last one to have had focus
4. Type the command’s keyboard shortcut

### Expected behavior
B is copied.

### Behavior for @beefeater7 
A is copied.

### Behavior for myself
With CopyQ 13.0.0 on Windows 11 25H2, the Windows clipboard is emptied and errors are displayed in 2 notifications:
> Exception in "Copy test"
> Failed to copy to clipboard

> Command "Copy test"
> Exit code: 4 1. copy: 2. copyq()

**Comment by beefeater7** [comment]
```
CopyQ Clipboard Manager 13.0.0
Qt: 6.8.3
KNotifications: 6.12.0
Compiler: MSVC
Arch: x86_64-little_endian-llp64
OS: Windows 10 Version 22H2
```

**Comment by Chealer** [comment]
I reproduced the bug @beefeater7 initially reported several times, on both of my Windows 11 25H2 laptops, still with CopyQ 13.0.0. To clarify, the reproduction steps of this simple scenario are:
1. Minimize all windows (to show the desktop)
2. Select 1 or more files
3. Type the shortcut of a command which calls `copy()`.
----

Thank you @beefeater7, but you seem to be reporting 2 possibly distinct bugs here. If you want this ticket to track both, please retitle it more generally. I suggest:
> Commands calling copy(), when launched on files on the Windows desktop, copy the wrong file or generally fail with "Failed to copy to clipboard!" exception

The exception I get in the simple (original) scenario is:
```
Note [2026-01-25 15:55:42.554] <Server-43220>: Exception in command "copy() test": ScriptError: Failed to copy to clipboard!
Note [2026-01-25 15:55:42.554] <Server-43220>:    
Note [2026-01-25 15:55:42.554] <Server-43220>:    --- backtrace ---
Note [2026-01-25 15:55:42.554] <Server-43220>:    f@:1
Note [2026-01-25 15:55:42.554] <Server-43220>:    %entry@eval code:2
Note [2026-01-25 15:55:42.554] <Server-43220>:    f@:1
Note [2026-01-25 15:55:42.554] <Server-43220>:    eval:
Note [2026-01-25 15:55:42.554] <Server-43220>:    copy()
Note [2026-01-25 15:55:42.554] <Server-43220>:    
Note [2026-01-25 15:55:42.554] <Server-43220>:    eval
Note [2026-01-25 15:55:42.554] <Server-43220>:    --- end backtrace ---
Note [2026-01-25 15:55:42.574] <Server-43220>: Command “copy() test”
Note [2026-01-25 15:55:42.574] <Server-43220>:    Exit code: 4
Note [2026-01-25 15:55:42.574] <Server-43220>:    1. copyq: 
Note [2026-01-25 15:55:42.574] <Server-43220>:    2. copy()
```

Is that the same exception you get?
